### ISSUE TITLE
cy.wait() with alias cy.intercept() does not match child routes

### ISSUE THREAD

**Issue by DW8Reaper** [cypress-io/cypress#14239]
### Current behavior
when you use cy.intercept().as('some-alias') and then cy.wait() for that alias if a route has the same parent path as another alias the parent alias is always assigned and the child wait will fail

I believe this is similar to https://github.com/cypress-io/cypress/issues/9526 but my issue was happens reliably and only for parent/child routes but feel free to close this and answer there I posted comments there as well

### Desired behavior
child route should have its own alias recognized
### Test code to reproduce
I created a super simple reproduction on StackBlitz, you will have to run it locally though just do a ```npm start``` then ```npm run test```

https://stackblitz.com/edit/js-h7pfkz?file=package.json
### Versions
this happens on 6.0.1, 6.1.0 and 5.6.0 (with route2)

**Comment by bahmutov** [comment]
Look at the routes table in the command log. The first intercept (the parent) intercepts both calls

<img width="700" alt="Screen Shot 2020-12-18 at 12 36 02 PM" src="https://user-images.githubusercontent.com/2212006/102643735-adca9280-412d-11eb-82b7-4d78f550b355.png">

PS: you do not need to import anything from mocha when writing tests, functions like `beforeEach` are automatically global when running using Cypress

```
// do not import, not necessary
import { beforeEach } from "mocha";
```

**Comment by bahmutov** [comment]
Possible solutions:

Use regular expression when specifying the parent intercept

```js
cy.intercept("GET", /\/api\/parent\/a$/, "parent a data").as("parent-request");
```

<img width="803" alt="Screen Shot 2020-12-18 at 12 38 22 PM" src="https://user-images.githubusercontent.com/2212006/102643903-f8e4a580-412d-11eb-9534-f256b45141db.png">

Or specify the parent using the pathname

```js
cy.intercept({
      method: 'GET',
      pathname: '/api/parent/a'
    }, 'parent a data').as("parent-request");
```

<img width="811" alt="Screen Shot 2020-12-18 at 12 40 15 PM" src="https://user-images.githubusercontent.com/2212006/102644060-30535200-412e-11eb-860c-9df9472f7fc2.png">

For more discussions read https://glebbahmutov.com/blog/cypress-intercept-problems/

**Comment by DW8Reaper** [comment]
Thank you for the information I read through the page you reference and the documentation again and did not understand that the URL worked differently from how it did before with ```cy.route```. Unfortunately I have a couple thousand routes to update so I don't think the regex change is an option for me but the path option will work so thank you for your feedback.

As for your fist comment about the "parent" intercepting both in the log this is exactly the behavior I logged the issue for since it didn't match you cy.route worked before. Your suggestion of using path will resolve this for me, thanks for your help and I apologize if my original issue request seemed unappreciative or accusatory in any way that would not have been my intention. I do appreciate that you are providing this support basically for free.

**Comment by lapalb** [comment]
One of the way to fix it to declare child first and then parent

```js
cy.intercept("GET", "/api/parent/a/child", "child data").as(
      "child-request"
 );
cy.intercept("GET", "/api/parent/a", "parent a data").as("parent-request");
   
```

Definately not a good solution, but works

**Comment by lapalb** [comment]
Is this issue fixed in a later release? I am using 6.1
